datadog/resource_datadog_dashboard.go

```diff
@@ -7613,6 +7795,340 @@ func buildDatadogTraceServiceDefinition(terraformDefinition map[string]interface
+				datadogTagValues := make([]string, len(tagValuesList))
+				for k, tagValues := range tagValuesList {
+					datadogTagValues[k] = tagValues.(string)
+				}
+				datadogSplitVector.SetTagValues(datadogTagValues)
+			}
+			datadogStaticSplits[i] = append(datadogStaticSplits[i], datadogSplitVector)
+		}
+	}
+	return &datadogStaticSplits
+}
+
+func buildTerraformSplitGraphDefinition(datadogDefinition *datadogV1.SplitGraphWidgetDefinition) map[string]interface{} {
+	terraformDefinition := map[string]interface{}{}
+	// Required params
+	if v, ok := datadogDefinition.GetSourceWidgetDefinitionOk(); ok {
+		terraformDefinition["source_widget_definition"] = []map[string]interface{}{buildTerraformSourceWidgetDefinition(v)}
+	}
+	if v, ok := datadogDefinition.GetSizeOk(); ok {
+		terraformDefinition["size"] = v
+	}
+	if v, ok := datadogDefinition.GetSplitConfigOk(); ok {
+		terraformDefinition["split_config"] = []map[string]interface{}{*buildTerraformSplitConfig(v)}
+	}
+	// Optional params
+	if v, ok := datadogDefinition.GetHasUniformYAxesOk(); ok {
+		terraformDefinition["has_uniform_y_axes"] = *v
+	}
+	if v, ok := datadogDefinition.GetTitleOk(); ok {
+		terraformDefinition["title"] = *v
+	}
+	if v, ok := datadogDefinition.GetTimeOk(); ok {
+		terraformDefinition["live_span"] = v.GetLiveSpan()
+	}
+
+	return terraformDefinition
+}
+
+func buildTerraformSplitConfig(datadogSplitConfig *datadogV1.SplitConfig) *map[string]interface{} {
+	terraformSplitConfig := map[string]interface{}{}
+	if v, ok := datadogSplitConfig.GetSplitDimensionsOk(); ok {
+		terraformOneGraphPer := map[string]interface{}{}
+		datadogSplitDimensions := *v
+		terraformOneGraphPer["one_graph_per"] = datadogSplitDimensions[0].OneGraphPer
+		terraformSplitConfig["split_dimensions"] = []map[string]interface{}{terraformOneGraphPer}
+	}
+	if v, ok := datadogSplitConfig.GetLimitOk(); ok {
+		terraformSplitConfig["limit"] = *v
+	}
+	if datadogSort, ok := datadogSplitConfig.GetSortOk(); ok {
+		terraformSortList := make([]map[string]interface{}, 1)
+		terraformSort := map[string]interface{}{}
+		terraformSortComputeList := make([]map[string]interface{}, 1)
+
+		terraformSortCompute := map[string]interface{}{}
+		terraformSortCompute["aggregation"] = datadogSort.Compute.Aggregation
+		terraformSortCompute["metric"] = datadogSort.Compute.Metric
+		terraformSortComputeList[0] = terraformSortCompute
+
+		terraformSort["order"] = datadogSort.Order
+		terraformSort["compute"] = terraformSortComputeList
+
+		terraformSortList[0] = terraformSort
+		terraformSplitConfig["sort"] = terraformSortList
+	}
+	if v, ok := datadogSplitConfig.GetStaticSplitsOk(); ok {
+		terraformSplitConfig["static_splits"] = buildTerraformStaticSplits(v)
+	}
+	return &terraformSplitConfig
+}
+
+// Build static splits for terraform from backend format
+func buildTerraformStaticSplits(datadogStaticSplits *[][]datadogV1.SplitVectorEntryItem) *[]interface{} {
```

Comment:
Terraform static splits have a slightly different shape than datadog static splits, because of terraform limitations.
In order to support array of arrays we add another level of `split_vector`